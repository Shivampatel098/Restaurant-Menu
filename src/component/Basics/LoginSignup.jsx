import React, { useState } from 'react'
import "./LoginSignup.css"

import userIcon from './Icons/Personlogo.png'
import eamilIcon from './Icons/Emaillogo.png'
import passwordIcon from './Icons/Passwordlogo.png'
import { json, useNavigate } from 'react-router-dom'
const LoginSignup = () => {
    const [action,setAction] =useState("Sign Up");
    const [name,setName] =useState("");
    const [email,setEmail] =useState("");
    const [password,setPassword] =useState("");
    
    const navigate = useNavigate();

    const validate = ()=>
    {
        let isProceed =true;
        let errmsg = 'Please enter the values ';
        if(action ==="Sign Up")
        {
            if(name === null || name === '')
            {
                isProceed=false;
                errmsg += 'Username';
            }
        }
        
        if(email === null || email === '')
        {
            isProceed=false;
            errmsg += 'Email Id';
        }
        if(password === null || password === '')
        {
            isProceed=false;
            errmsg += 'Password';
        }
        if(!isProceed){
            console.log(errmsg);
        }
        else{
            if(/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/.test(email))
            {

            }
            else{
                isProceed =false;
                console.log("Please enter valid email id.")
            }
        }
        return isProceed;
    }
    const handleSignup = (e) => {
        e.preventDefault();
        if(validate()){
        
        let register={name,email,password};
        fetch(" http://localhost:8000/user",{
            method:"POST",
            headers:{'content-type':'application/json'},
            body:JSON.stringify(register)

        }).then((res)=>{
            console.log("Resistered successfully");
            
// set some message SUCCESSso that the user can know
        }).catch((err)=>{
            console.log("Registration failed.");
        })
    }
}
    const handleLogin = (e) => {
        e.preventDefault();
        if(validate())
        {
            // fetch("http://localhost:8000/user/" + email).then((res) => {
            //     return res.json();
            // }).then((resp) => {
            //     // console.log(resp);
            //     if(Object.keys(resp).length === 0)
            //     {
            //         console.log("Please enter valid EmailId");
            //     }
            //     else{
            //         if(resp.password === password)
            //         {
            //             navigate('/Restaurant');
            //         }
            //         else{
            //             console.log("Please enter valid password");
            //         }
            //     }
                
            // }).catch((err)=>{
            //     console.log("Login Unsuccessfull");
            // })
        }
        
    }

    const handleActionChange = (newAction) =>{
        setName("");
        setEmail("");
        setPassword("");
        setAction(newAction);
    }
  return (
    <div className='container'>
        <div className="header">
            <div className="text">{action}</div>
            <div className="underline"></div>
        </div>
        <div className="inputs">
            {action === "Login"?<div></div>:<div className="input">
                <img src={userIcon} alt="user icon" />
                <input value={name} onChange={e=>setName(e.target.value)} type="text" placeholder='Name'/>
            </div>}
            
            <div className="input">
                <img src={eamilIcon} alt="email icon" />
                <input value={email} onChange={e=>setEmail(e.target.value)} type="email" placeholder='Email ID' />
            </div>
            <div className="input">
                <img src={passwordIcon} alt="password icon" />
                <input value={password} onChange={e=>setPassword(e.target.value)} type="password" placeholder='Password' />
            </div>
        </div>
        {action ==="Sign Up"?<div></div>:<div className="forgot-password">Lost Password? <span>Click Here</span></div>}
        <div className="submit-container">
            <button type="submit" role="signup-button" className={action === "Login"?"submit gray":"submit"} onClick={(e) => {
                    e.preventDefault();
                    handleActionChange("Sign Up");
                    handleSignup(e);
                }}>Sign Up</button>
            <button type="submit" className={action === "Sign Up"?"submit gray":"submit"} onClick={(e) => {
                    e.preventDefault();
                    handleActionChange("Login");
                    handleLogin(e);
                }}>Login</button>
        </div>

        
    </div>
  )
}
export default LoginSignup
