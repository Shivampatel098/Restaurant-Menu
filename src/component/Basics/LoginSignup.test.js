/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import { render, fireEvent } from '@testing-library/react'; // Import the necessary testing functions
import { MemoryRouter } from 'react-router-dom';
import LoginSignup from './LoginSignup';

describe('LoginSignup Component', () => {
  it('Renders the Login form by default', () => {
    const component = (
      <MemoryRouter>
        <LoginSignup />
      </MemoryRouter>
    );    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it('Switches to the Sign Up form when "Sign Up" button is clicked', () => {
    const component = (
      <MemoryRouter>
        <LoginSignup />
      </MemoryRouter>
    );
    const { container, getByRole } = render(component);
  
    // Find and click the "Sign Up" button by its role
    const signUpButton = getByRole('signup-button');
    fireEvent.click(signUpButton);

    expect(container).toMatchSnapshot();
  });

  it('Validates form inputs and displays error messages', () => {
    const component = (
      <MemoryRouter>
        <LoginSignup />
      </MemoryRouter>
    );    const { container, getByText, getByRole, getByPlaceholderText } = render(component);

    // Find and click the "Sign Up" button
    const signUpButton = getByRole('signup-button');
    fireEvent.click(signUpButton);

    // Find input fields
    const nameInput = getByPlaceholderText('Name');
    const emailInput = getByPlaceholderText('Email ID');
    const passwordInput = getByPlaceholderText('Password');

    // Find the submit button
    const submitButton = getByRole('signup-button');

    // Click the submit button without filling in any fields
    fireEvent.click(submitButton);
    expect(container).toMatchSnapshot();

    // Fill in an invalid email and click the submit button
    fireEvent.change(emailInput, { target: { value: 'invalid-email' } });
    fireEvent.click(submitButton);
    expect(container).toMatchSnapshot();

    // Fill in a valid email and click the submit button
    fireEvent.change(emailInput, { target: { value: 'valid-email@example.com' } });
    fireEvent.click(submitButton);
    expect(container).toMatchSnapshot();
  });
});
