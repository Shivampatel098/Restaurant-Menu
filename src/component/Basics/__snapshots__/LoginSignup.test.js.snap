// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSignup Component Renders the Login form by default 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="text"
      >
        Sign Up
      </div>
      <div
        class="underline"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="input"
      >
        <img
          alt="user icon"
          src="Personlogo.png"
        />
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="email icon"
          src="Emaillogo.png"
        />
        <input
          placeholder="Email ID"
          type="email"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="password icon"
          src="Passwordlogo.png"
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div />
    <div
      class="submit-container"
    >
      <button
        class="submit"
        role="signup-button"
        type="submit"
      >
        Sign Up
      </button>
      <button
        class="submit gray"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginSignup Component Switches to the Sign Up form when "Sign Up" button is clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="text"
      >
        Sign Up
      </div>
      <div
        class="underline"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="input"
      >
        <img
          alt="user icon"
          src="Personlogo.png"
        />
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="email icon"
          src="Emaillogo.png"
        />
        <input
          placeholder="Email ID"
          type="email"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="password icon"
          src="Passwordlogo.png"
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div />
    <div
      class="submit-container"
    >
      <button
        class="submit"
        role="signup-button"
        type="submit"
      >
        Sign Up
      </button>
      <button
        class="submit gray"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginSignup Component Validates form inputs and displays error messages 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="text"
      >
        Sign Up
      </div>
      <div
        class="underline"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="input"
      >
        <img
          alt="user icon"
          src="Personlogo.png"
        />
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="email icon"
          src="Emaillogo.png"
        />
        <input
          placeholder="Email ID"
          type="email"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="password icon"
          src="Passwordlogo.png"
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div />
    <div
      class="submit-container"
    >
      <button
        class="submit"
        role="signup-button"
        type="submit"
      >
        Sign Up
      </button>
      <button
        class="submit gray"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginSignup Component Validates form inputs and displays error messages 2`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="text"
      >
        Sign Up
      </div>
      <div
        class="underline"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="input"
      >
        <img
          alt="user icon"
          src="Personlogo.png"
        />
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="email icon"
          src="Emaillogo.png"
        />
        <input
          placeholder="Email ID"
          type="email"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="password icon"
          src="Passwordlogo.png"
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div />
    <div
      class="submit-container"
    >
      <button
        class="submit"
        role="signup-button"
        type="submit"
      >
        Sign Up
      </button>
      <button
        class="submit gray"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginSignup Component Validates form inputs and displays error messages 3`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="text"
      >
        Sign Up
      </div>
      <div
        class="underline"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="input"
      >
        <img
          alt="user icon"
          src="Personlogo.png"
        />
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="email icon"
          src="Emaillogo.png"
        />
        <input
          placeholder="Email ID"
          type="email"
          value=""
        />
      </div>
      <div
        class="input"
      >
        <img
          alt="password icon"
          src="Passwordlogo.png"
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div />
    <div
      class="submit-container"
    >
      <button
        class="submit"
        role="signup-button"
        type="submit"
      >
        Sign Up
      </button>
      <button
        class="submit gray"
        type="submit"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
